### ISSUE TITLE
Make truffle exec compile contracts before running scripts

### ISSUE THREAD

**Issue by maraoz** [trufflesuite/truffle#908]
This is a suggestion, but I don't see the advantage of not compiling the contracts, and it makes `truffle exec` less useful if you have to compile them manually before each run. 

I always end up doing: 
```
npx truffle compile && npx truffle exec myScript.js
```

**Comment by cgewecke** [comment]
Thanks @maraoz, this seems right. 

After some discussion with @gnidan, have implemented this as an option in [truffle-core 132](https://github.com/trufflesuite/truffle-core/pull/132) e.g:
```
truffle exec [-c | --compile] <script.js>
```
`compile` overwrites the artifact bytecode so it might be safest to require the user express a clear intent here - it's a change in default behavior that saves to file. 

Should be in the next patch release, thanks again.

**Comment by cgewecke** [comment]
Published with `4.1.8`
